src/app/result.ts

```diff
@@ -15,25 +22,26 @@ export class AsyncResult {
     this.taskId = taskId;
     this.backend = backend;
     this.result = null;
+    this._promise = null;
   }
 
   /**
    * @method AsyncResult#get
    * @returns {Promise}
    */
   get(timeout?: number): Promise<any> {
-    return new Promise((resolve, reject) => {
-      if (this.result) {
-        resolve(this.result);
-      }
+    if (this._promise) {
```

Comment:
I've gone into more detail about this change in the comment message for https://github.com/actumn/celery.node/pull/25/commits/d69de37733f567b5615056b9414aa06afd835387